### PR TITLE
Add completion support within function call expressions

### LINKED ISSUE #28354: Gives context irrelevant suggestions
See the image for more details. Just as I type "http:" I get this.

Screen Shot 2021-02-01 at 9 49 20 PM

### PR DESCRIPTION
**Pull request by nadeeshaan**
1. Add completion support within the function call expression
2. Avoid suggestions within the string literals

Fixes #28354

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by IMS94 — APPROVED** [review]
LGTM!

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BasicLiteralNodeContext.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (prabushi) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BasicLiteralNodeContext.java:2]: Shall we update the license headers?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FunctionCallExpressionNodeContext.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (dulajdilshan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FunctionCallExpressionNodeContext.java:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```

Comment (dulajdilshan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FunctionCallExpressionNodeContext.java:2]: Shall we update this?

**Comment by pubudu91** [comment]
Does this support the following?
```ballerina
function testFunction() {
    int valueA = 123;
    string s = "foo";
    testFunctionWithParams(valueA, <cursor>);
}
```
`s` shouldn't be suggested as a completion for the given cursor position. If this is supported, shall we add a test case for that as well?

**Comment by nadeeshaan** [comment]
> Does this support the following?
> 
> ```ballerina
> function testFunction() {
>     int valueA = 123;
>     string s = "foo";
>     testFunctionWithParams(valueA, <cursor>);
> }
> ```
> 
> `s` shouldn't be suggested as a completion for the given cursor position. If this is supported, shall we add a test case for that as well?
> 
> Other than that, it LGTM.

I think we do need to suggest the variable `s` here. This is because user can invoke methods such as `length`, `size`, over the variable. Since we suggest. One option is to like lower  the priority of the particular item.

**Review by rasika — APPROVED** [review]

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**